geps/gep-1713.md

```diff
@@ -0,0 +1,103 @@
+# GEP-1713: Standard Mechanism to Merge Multiple Gateways
+
+* Issue: [#1713/kubernetes-sigs/gateway-api/issues/1713)
+* Status: Provisional
+
+(See status definitions [here](overview.md#status).)
+
+## TLDR
+
+The Gateway Resource is a contention point since it is the only place to attach listeners with certificates. We propose a mechanism to allow distinct Gateway resources to be logically merged.
```

Comment:
I'm not sure I see how this will address issues with attaching certificates.  Can you elaborate a bit more on what the core issue is with certificate attachment only existing in the Gateway, the personas might need to attach certificates, and update the example to include certificate attachment?

Reply:
The root issue here is that the Knative control planes can't set a certificate for an HTTPRoute. There are valid use cases for this - ie. HTTP01 challenges produce a non-wildcard cert. 

For Knative we produce a certificate for each Knative Service with the host name of the form `{service-name}.{namespace}.some-operator-defined.tld`

In the GatewayAPI world we need to attach this certificate to the Gateway resource. But there's a disconnect between the lifecycle of the certificate and the Gateway. The cert lifecycle aligns more with the HTTPRoute (ie. deleting the Knative Service should delete this certificate). We also support HA controllers and each of them manipulating a single resource will cause 409 conflicts in the K8s API.

More details are here: https://github.com/kubernetes-sigs/gateway-api/discussions/1248

If you think it's worth revisiting TLS setting on an HTTPRoute I'd be open to starting that discussion.

Reply:
> If you think it's worth revisiting TLS setting on an HTTPRoute I'd be open to starting that discussion.
Asked in slack - https://kubernetes.slack.com/archives/CR0H13KGA/p1679510731917299